### PR TITLE
[Android] SearchHandler - added focus/unfocus support

### LINKED ISSUE #24670: [Android] SearchHandler.Focused event never fires
### Description

The Focused event for Shell.SearchHandler does not fire on selecting it, even though the keyboard shows up.

### Steps to Reproduce

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="MauiApp1.NewPage1"
             Title="NewPage1">
    <Shell.SearchHandler>
        <SearchHandler
            Focused="SearchHandler_Focused"
            Placeholder="Click to focus.." />
    </Shell.SearchHandler>
    <VerticalStackLayout>
        <Label 
            Text="Welcome to .NET MAUI!"
            VerticalOptions="Center" 
            HorizontalOptions="Center" />
    </VerticalStackLayout>
</ContentPage>
```

```cs
namespace MauiApp1;

public partial class NewPage1 : ContentPage
{
    public NewPage1()
    {
        InitializeComponent();
    }

    private void SearchHandler_Focused(object sender, EventArgs e)
    {
        // Set a breakpoint
    }
}
```

SearchHandler_Focused is never invoked on tapping on the SearchHandler.

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.80 SR8

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

No workaround.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Description of Change

Added a missing support for focus/unfocus events. Implemented it similar to how it has been done for iOS 
(https://github.dev/kubaflo/maui/blob/8921d3bfab72dc78aba940020a6b2ea06a708408/src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/SearchHandlerAppearanceTracker.cs#L32-L49)

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/24670

<video src="https://github.com/user-attachments/assets/47372cf0-4bfa-4d55-bb05-02b99157c1a6" width="300px"/>

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24670.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		public override string Issue => "SearchHandler.Focused event never fires";
+
+		public Issue24670(TestDevice testDevice) : base(testDevice)
+		{
+		}
+
+		[Test]
+		[Category(UITestCategories.Shell)]
+		[Category(UITestCategories.SearchBar)]
+		public void SearchHandlerFocusAndUnfocusEventsShouldWork()
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24670.cs:18]: This test is failing on Android. Maybe need to use `WaitForElement` before Click the Button.

```
System.NullReferenceException : Object reference not set to an instance of an object.
 at UITest.Appium.HelperExtensions.Click(IUIElement element) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 154
   at UITest.Appium.HelperExtensions.Click(IApp app, String element) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 34
   at Microsoft.Maui.TestCases.Tests.Issues.Issue24670.SearchHandlerFocusAndUnfocusEventsShouldWork() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24670.cs:line 20
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/SearchHandlerAppearanceTracker.cs [hunk 2]
```diff
@@ -83,6 +84,11 @@ protected virtual void SearchHandlerPropertyChanged(object sender, System.Compon
 			}
 		}
 
+		void EditTextFocusChange(object s, AView.FocusChangeEventArgs args)
+		{
+			_searchHandler.SetIsFocused(_editText.IsFocused);
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/SearchHandlerAppearanceTracker.cs:89]: Mnnn, here, should show the keyboard if `IsFocused` is true and hide the keyboard in other case.
Have extension methods to do it in the ContextExtensions class: https://github.com/dotnet/maui/blob/d91ba8e19ecae1d03f5a84a454c113582bf2c749/src/Core/src/Platform/Android/ContextExtensions.cs#L80

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/SearchHandlerAppearanceTracker.cs [hunk 3]
```diff
@@ -83,6 +84,16 @@ protected virtual void SearchHandlerPropertyChanged(object sender, System.Compon
...
+
+			if (_editText.IsFocused)
```
Comment (PureWeen) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/SearchHandlerAppearanceTracker.cs:91]: Why do we need to add this behavior? 

The EditText should just open the keyboard itself.

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/SearchHandlerAppearanceTracker.cs (hunk 3 above)
Comment (kubaflo) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/SearchHandlerAppearanceTracker.cs]: Right! I removed it

**Review by PureWeen — APPROVED** [review]

**Pull request merged**